rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -431,6 +447,19 @@ def parse_message_string(pkg_name, msg_name, message_string):
                     line=line, pkg=pkg_name, msg=msg_name, err=err,
                 ))
                 raise
+
+            # check for field name collision with action implicit parameters
+            # (e.g. 2 or more occurrences of a field_name contained in ACTION_IMPLICIT_FIELDS,
+            # including the implicit one)
+            if field_name in ACTION_IMPLICIT_FIELDS and action_fields[field_name] >= 1:
```

Comment:
@jacobperron It will trigger only if a field named with the same name as one implicit field is in the spec and there are at least two of them (replacing a `ValueType` error in this case), to make sure it triggers only when there's duplication, so if you have a field named `uuid` used in your msg definition it will not trigger. Should I go with the invalid fields route, then?

Reply:
Thanks for the clarification. I think it is fine to leave it as-is.